ext/intl/listformatter/listformatter_class.c

```diff
@@ -0,0 +1,217 @@
+PHP_METHOD(IntlListFormatter, format)
+{
+    ListFormatter_object *obj = Z_INTL_LISTFORMATTER_P(ZEND_THIS);
+    zval *strings;
+
+    ZEND_PARSE_PARAMETERS_START(1, 1)
+        Z_PARAM_ARRAY(strings)
+    ZEND_PARSE_PARAMETERS_END();
+
+    if (!LISTFORMATTER_OBJECT(obj)) {
+        intl_error_set(NULL, U_ILLEGAL_ARGUMENT_ERROR, "ListFormatter not properly constructed", 0);
+        RETURN_FALSE;
+    }
+
+    HashTable *ht = Z_ARRVAL_P(strings);
+    uint32_t count = zend_array_count(ht);
+    if (count == 0) {
+        RETURN_EMPTY_STRING();
+    }
+
+    const UChar **items = (const UChar **)safe_emalloc(count, sizeof(const UChar *), 0);
+    int32_t *itemLengths = (int32_t *)safe_emalloc(count, sizeof(int32_t), 0);
+    uint32_t i = 0;
+    zval *val;
+
+    ZEND_HASH_FOREACH_VAL(ht, val) {
+        zend_string *str_val;
+        
+        str_val = zval_get_string(val);
+        
+        // Convert PHP string to UTF-16
+        UChar *ustr = NULL;
+        int32_t ustr_len = 0;
+        UErrorCode status = U_ZERO_ERROR;
+        
+        intl_convert_utf8_to_utf16(&ustr, &ustr_len, ZSTR_VAL(str_val), ZSTR_LEN(str_val), &status);
+        if (U_FAILURE(status)) {
+            efree(items);
+            efree(itemLengths);
+            zend_string_release(str_val);
+            intl_error_set(NULL, status, "Failed to convert string to UTF-16", 0);
+            RETURN_FALSE;
+        }
+        
+        items[i] = ustr;
+        itemLengths[i] = ustr_len;
+        zend_string_release(str_val);
+        i++;
+    } ZEND_HASH_FOREACH_END();
+
+    UErrorCode status = U_ZERO_ERROR;
+    int32_t resultLength;
+    UChar *result = NULL;
+
+    resultLength = ulistfmt_format(LISTFORMATTER_OBJECT(obj), items, itemLengths, count, NULL, 0, &status);
+
+    if (U_FAILURE(status) && status != U_BUFFER_OVERFLOW_ERROR) {
+        intl_error_set(NULL, status, "Failed to format list", 0);
+        RETURN_FALSE;
+    }
+
+    // Allocate buffer and try again
+    status = U_ZERO_ERROR;
+    result = (UChar *)emalloc((resultLength + 1) * sizeof(UChar));
+    ulistfmt_format(LISTFORMATTER_OBJECT(obj), items, itemLengths, count, result, resultLength, &status);
+
+    // Clean up input strings
+    for (i = 0; i < count; i++) {
+        efree((void *)items[i]);
+    }
+    efree(items);
+    efree(itemLengths);
+
+    if (U_FAILURE(status)) {
+        if (result) {
+            efree(result);
+        }
+        intl_error_set(NULL, status, "Failed to format list", 0);
+        RETURN_FALSE;
+    }
+
+    // Convert result back to UTF-8
+    zend_string *ret = intl_convert_utf16_to_utf8(result, resultLength, &status);
+    efree(result);
+    
+    if (!ret) {
+        intl_error_set(NULL, status, "Failed to convert result to UTF-8", 0);
+        RETURN_FALSE;
+    }
+
+    RETURN_STR(ret);
+}
+/* }}} */
+
+/* {{{ listformatter_getErrorCode */
+PHP_METHOD(IntlListFormatter, getErrorCode)
+{
+    ListFormatter_object *obj = Z_INTL_LISTFORMATTER_P(ZEND_THIS);
```

Comment:
The build errors are legit, you probably need `ZEND_PARSE_PARAMETERS_NONE` macro.

Reply:
Thanks! I pushed a commit that adds it for both methods.